### ISSUE TITLE
Fix `declaration-block-no-duplicate-properties` autofix for `!important`

### ISSUE THREAD

**Issue by sidx1024** [stylelint/stylelint#6523]
### What steps are needed to reproduce the bug?

1. Enable `declaration-block-no-duplicate-properties` rule
2. Run `stylelint --fix` against a file like this:
```scss
.box {
  display: flex !important;
  display: inline-flex;
}
```

The css fixed by stylelint becomes:
```scss
.box {
  display: inline-flex !important;
  display: inline-flex;
}
```
The fix is incorrect because before fixing the css the effective value for display would be `flex` and after the fix it is `inline-flex`.

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "stylelint-scss",
  ],
  "rules": {
    "declaration-block-no-duplicate-properties": true
  }
}
```

### How did you run Stylelint?

stylelint file.scss --fix

### Which version of Stylelint are you using?

13.12.0

### What did you expect to happen?

The expected file after auto fix should be:
```scss
.box {
  display: flex !important;
}
```
If the case is difficult to handle the issue should not be auto-fixed.

### What actually happened?

The issue was auto-fixed and the file after the fix became:
```scss
.box {
  display: inline-flex !important;
  display: inline-flex;
}
```

### Proposal to fix the bug

1. Avoid auto-fixing if one of the duplicates have `!important`.
2. Apply the effective value for the property when `!important` is present.

**Comment by ybiquitous** [comment]
@sidx1024 Thanks for the report. As you commented, the current `declaration-block-no-duplicate-properties` rule doesn't consider `!important`, and I agree that the rule auto-fix should remain only one effective property.

I guess there are the following cases for `!important`:

### Case 1

When there is one `!important`:

```css
a {
  color: red !important;
  color: green;
}
```

should become ⬇️ 

```css
a {
  color: red !important;
}
```

### Case 2

When there are multiple `!important`:

```css
a {
  color: red !important;
  color: green !important;
}
```

should become ⬇️ 

```css
a {
  color: green !important;
}
```

In other words, only the last `!important` should be kept.

What do you think?

**Comment by jeddy3** [comment]
> In other words, only the last `!important` should be kept.

SGTM.

I've labelled the issue as ready to implement. @sidx1024 Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by sidx1024** [comment]
Sure. Raised a PR https://github.com/stylelint/stylelint/pull/6528
